Sources/mpc-core-kit-swift/MpcCoreKitSwift.swift

```diff
@@ -289,19 +270,29 @@ public class MpcCoreKit {
 
         let _ = try await tkey.reconstruct()
 
+        // backup metadata share using factorKey
+        // finding device share index
+        var shareIndexes = try tkey.get_shares_indexes()
+        shareIndexes.removeAll(where: { $0 == "1" })
+
         // TSS Module Initialize - create default tag
         // generate factor key or use oauthkey hash as factor
         let factorKey: String
         let descriptionTypeModule: FactorType
-        
+
+        let hashFactorKey = try Utilities.getHashedPrivateKey(postboxKey: oAuthKey!, clientID: option.web3AuthClientId)
+
         if option.disableHashFactor == false {
-            factorKey = try Utilities.getHashedPrivateKey(postboxKey: oauthKey!, clientID: option.Web3AuthClientId)
+            factorKey = hashFactorKey
             descriptionTypeModule = FactorType.HashedShare
 
         } else {
             // random generate
             factorKey = try curveSecp256k1.SecretKey().serialize()
```

Comment:
in web its being done using a hashedNonce input from constructor

Reply:
@AyushBherwani1998

Reply:
Done